automatic_push/utils.py

```diff
@@ -0,0 +1,166 @@
+            os.mknod(git_config_lock_file)
+
+        # git fetch --all
+        for remote in main_repo.repo.remotes:
+            info = remote.fetch('main')[0]
+            logger.info(f'Remote: {remote.name} - Commit: {info.commit}')
+
+        # git pull origin main
+        origin.pull('main')
+
+        # git push fork main
+        push_info = fork.push('main')[0]
+        logger.info(f'Push info: {push_info.summary}')
+        if 'non-fast-forward' in push_info.summary:
+            logger.warning('yang-catalog/yang repo might not be up-to-date')
+    except GitCommandError:
+        logger.exception('yang-catalog/yang repo might not be up-to-date')
+
+
+def download_draft_modules_content(experimental_path: str, config: ConfigParser, logger: logging.Logger):
+    """
+    Loop through download links for each module found in IETFDraft.json and try to get their content.
+
+    Arguments:
+        :param experimental_path    (str) full path to the directory with cloned experimental modules
+        :param config               (configparser.ConfigParser) instance of ConfigParser class
+        :param logger               (logging.Logger) formated logger with the specified name
+    """
+    ietf_draft_url = config.get('Web-Section', 'ietf-draft-private-url')
+    my_uri = config.get('Web-Section', 'my-uri')
+    domain_prefix = config.get('Web-Section', 'domain-prefix')
+    ietf_draft_json = {}
+    response = requests.get(ietf_draft_url)
+    try:
+        ietf_draft_json = response.json()
+    except json.decoder.JSONDecodeError:
+        logger.error(f'Unable to get content of {os.path.basename(ietf_draft_url)} file')
+    for key in ietf_draft_json:
+        file_path = os.path.join(experimental_path, key)
+        yang_download_link = ietf_draft_json[key]['compilation_metadata'][2].split('href="')[1].split('">Download')[0]
+        yang_download_link = yang_download_link.replace(domain_prefix, my_uri)
+        try:
+            file_content_response = requests.get(yang_download_link)
+        except ConnectionError:
+            logger.error(f'Unable to retrieve content of: {key} - {yang_download_link}')
+            continue
+        if 'text/html' in file_content_response.headers['content-type']:
+            logger.error(f'The content of "{key}" file is a broken html, download link: {yang_download_link}')
+            if not os.path.exists(file_path):
+                continue
+            with open(file_path, 'r') as possibly_broken_module:
+                lines = possibly_broken_module.readlines()
+                module_is_broken = '<html>' in lines[1] and '</html>' in lines[-1]
+            if module_is_broken:
+                logger.info(f'Deleted the file because of broken content: {key} - {yang_download_link}')
+                os.remove(file_path)
+            continue
+        with open(file_path, 'w') as yang_file:
+            yang_file.write(file_content_response.text)
+
+
+def push_untracked_files(
```

Comment:
Same for pushing, it would be better if all the functionality related to the manipulation with GitHub will be in same file.